### PR TITLE
feat: implement Media Chrome HTML based theme 

### PR DESCRIPTION
**Pull request by luwes**
this adds a working version of the Mux player theme as a Media Chrome theme.

MC templating concepts added:

- `{{ mustache }}` like vars
- inner `<template>` elements called directives that can be used for things like conditionals, loops, partials, etc
- `{{ myVar | filter }}` pipe filters on the vars that act as functions on the value stored in the variable
- `{{ myVar ?? 'fallback' }}` coalesce operator which can provide a default value
- `{{>MyPlayButton param="test" another="lala"}}` param passing to partials

todo: 
- [x] take in account escaping of characters in variables and how this should act in partials maybe need a new syntax for partials like `{{>MyPartialInHTML}}`
- [x] delete the old code of the theme
- [x] remove the `@github/template-parts` dependency. 
  it's probably best we maintain our own version and we require the InnerTemplatePart's so 
- [x] move `MediaThemeTemplate` to MediaChrome and formalize there more
- [x] upgrade Media Chrome to stable version

### CODE REVIEW

File: packages/mux-player/src/media-theme-mux.html
Comment (cjpillsbury) [inline packages/mux-player/src/media-theme-mux.html:291]: question(non-blocking): Any reason we couldn't declare the partials above or below all the other elements? If not, that might be a good convention for this use case ("reusable templated component renderings")

File: packages/mux-player/src/media-theme-mux.html
Comment (cjpillsbury) [inline packages/mux-player/src/media-theme-mux.html:348]: suggestion(non-blocking): It would be unfortunate if we lost the ability for designers to create svgs that they could just use with the src. This would be a good thing to follow up on so we don't have to again inline everything. The same applies to the styles.css change imo.

File: packages/mux-player/test/web-test-runner.config.mjs
Comment (cjpillsbury) [inline packages/mux-player/test/web-test-runner.config.mjs:22]: suggestion(non-blocking): Since this isn't "standard html," (aka it is intended to be used as a template) and per some of the other build-level callouts (css & svg inclusions), we should consider making this extension something like `.html.templ` (or whatever feels right that isn't simply `.html`) so we can distinctly target it for build tools going forward.

File: scripts/esbuilder/esbuilder.js
Comment (cjpillsbury) [inline scripts/esbuilder/esbuilder.js:58]: same as above

File: yarn.lock
Comment (cjpillsbury) [inline yarn.lock:10636]: issue(blocking): we should hold off on merging this until we get an official release out with your changes

File: packages/mux-player/src/template.ts
Comment (cjpillsbury) [inline packages/mux-player/src/template.ts:16]: question(blocking): Is this needed? From what I can tell, we're

1. pulling in an html template file as a string
2. using JS to create a template element to then set the innerHTML to that string
3. using a clone of that template instance's content as the thing to render into the template.

Couldn't we skip all of that and simply do something like:

```js
const staticTemplateStrPart = `
<style>
  ${cssStr}
</style>
${muxTheme}`;

export const template = (props: MuxTemplateProps) => html`
${staticTemplateStrPart}
${content(props)}
`;
```

? If not, can you briefly explain why not?

File: packages/mux-player/src/media-theme-mux.html
Comment (luwes) [reply packages/mux-player/src/media-theme-mux.html:291]: this felt most natural to me, above the layout templates where we use them

File: packages/mux-player/src/html.ts
Comment (cjpillsbury) [inline packages/mux-player/src/html.ts:3]: 😎

File: packages/mux-player/src/html.ts
Comment (cjpillsbury) [inline packages/mux-player/src/html.ts:118]: question(non-blocking): Why did this have to be converted to `any`? Is it possible to avoid this?

File: packages/mux-player/test/web-test-runner.config.mjs
Comment (luwes) [reply packages/mux-player/test/web-test-runner.config.mjs:22]: I don't think that's needed here, it's valid HTML. it being the html extension right away gets the right syntax in the editor. if we had many html files we import maybe but this is really a rare case

File: packages/mux-player/src/html.ts
Comment (cjpillsbury) [inline packages/mux-player/src/html.ts:153]: question(blocking): am I missing something or did we loose our `stringsCache` and any equivalent? Could this contribute to issues we've been discussing re:rendered template GC & duplication?

File: packages/mux-player/test/web-test-runner.config.mjs
Comment (cjpillsbury) [reply packages/mux-player/test/web-test-runner.config.mjs:22]: right, this was related to being able to externalize e.g. the svg files and the css file

**Review by cjpillsbury — CHANGES_REQUESTED** [review]
a few callouts, but this is looking 🔥

File: packages/mux-player/src/media-theme-mux.html
Comment (cjpillsbury) [reply packages/mux-player/src/media-theme-mux.html:291]: cool that makes sense to me, even if not my S1. Might be good to get @dylanjha and @gkatsev (and others') S1s here as well, even if I don't think we should block merging this regardless of folks' initial/gut thoughts.

File: packages/mux-player/src/html.ts
Comment (luwes) [reply packages/mux-player/src/html.ts:153]: this logic was to add unsafe statics like a dynamic tag name but this isn't needed anymore because custom themes will be identified by an id attribute. not custom element tags anymore.

I don't think it will have any effect on the issue but lets see. it can only be positive.

File: packages/mux-player/src/template.ts
Comment (luwes) [reply packages/mux-player/src/template.ts:16]: the html string is automatically escaped so it would just be text. we need to take advantage of the processing of this part https://github.com/muxinc/elements/blob/f24efa9501252d1d7385d359bc06475ecdd7f236/packages/mux-player/src/html.ts#L56-L62

File: packages/mux-player/src/html.ts
Comment (luwes) [reply packages/mux-player/src/html.ts:118]: yes, there was always an annoying issue that I didn't want to spend time on. I'll check when moving this to MC

File: packages/mux-player/src/html.ts
Comment (cjpillsbury) [reply packages/mux-player/src/html.ts:153]: I'm not sure if I agree with "it can only be positive," if one of our concerns is GC of the `WeakMap` references (I thought you said the `stringsCache` was the thing guaranteeing a maintained reference), but we can move forward with this and find out.

File: packages/mux-player/src/template.ts
Comment (cjpillsbury) [reply packages/mux-player/src/template.ts:16]: ah we should try to resolve that (in a subsequent effort), since, in theory, the `template="templateId"` should do the processing, no?

**Review by cjpillsbury — APPROVED** [review]
Pre-approving with a few concerns we should monitor and pending the media chrome release + update of dependencies here.

**Pull request merged**
